Firestore/core/src/firebase/firestore/core/query.h

```diff
@@ -66,6 +66,8 @@ class Query {
   Query Filter(std::shared_ptr<core::Filter> filter) const;
 
  private:
+  friend bool operator==(const Query& lhs, const Query& rhs);
```

Comment:
Why private? Or does that not matter?

Reply:
I don't think it matters. I typically put it in the 'private' section since it's (a) an implementation detail, and (b) allows access to private stuff.

But it doubly doesn't matter in this case. The only reason the friend statement is there at all is to access filters_. However, Query (eventually) will have a filters() method which returns filters_. (The next commit adds this in and eliminates the friend statement.)

Reply:
FWIW, access specifiers don't affect `friend`.